src/shaders/endoftrackshader.cpp

```diff
@@ -15,12 +15,12 @@ void main()
 )--");
 
     QString fragmentShaderCode = QStringLiteral(R"--(
-uniform vec4 color;
-varying float vgradient;
+uniform highp vec4 color;
+varying highp float vgradient;
```

Comment:
`highp` support is optional in the fragment shader. `lowp` is sufficient for color afaik.
```suggestion
uniform lowp vec4 color;
varying mediump float vgradient;
```

Reply:
I have read that the hardware takes anyway the ALU it has which should be 32 bit for all floats. So I think there is no point in reducing it and risk regression compared to the OpenGL version that is 32 bit anyway. 

If we have a ES hardware that has only 24 bit floats among our targets, we need to reduce all float to mediump.

Reply:
I have no interest maintain two precisions for no reason. Do you have an example device, that is not supported right now? If not, let's learn from the user once rendering fails and save our developer resources here.

Reply:
there is no reason to maintain two precisions. We don't need `highp` for many of these fragment shader uniforms, so lets just choose the safer default.

Reply:
Yes that's why I picked highp. It works like OpenGL and we don't need to test two versions.

Reply:
We can change it at any time, no objections. It just requires tests, I don't want to do right now.

Reply:
Right. My argument is that we change it now because in the case where the GLES waveforms are broken because of `highp` people will just fallback to the cpu waveforms assuming their hardware is incapable, even though it technically would be.

Reply:
The alternative would to be `ifdef` it in the shader, that would likely be most robust, but creates a bunch of duplicated code across the shaders.

Reply:
The ifdef is already there in the Qt shader wrapper. 
`#define highp` 
Removing highp for the GL case.

Reply:
No I meant
```glsl
#ifndef GL_FRAGMENT_PRECISION_HIGH
#define highp mediump
#endif
```
in the fragment shader. Then the shader will use highp if available and fallback if otherwise.

Reply:
It turns out that it is already there behind the scenes: 
https://github.com/qt/qtbase/blob/737dd9595058a457fdd8f336ee8630cb1e2c57bb/src/opengl/qopenglshaderprogram.cpp#L445

Not sure if it is in action though.

Reply:
Ok weird, well I'll take the gamble then and hope it is in action on the platforms where it makes sense.